### PR TITLE
Add styling rules for subquery `EXISTS`

### PR DESCRIPTION
**Pull request by Mats-SX**
Restructure content into sections, based on type of formatting (rather than type of language construct).

### CODE REVIEW

File: docs/style-guide.adoc [hunk 1]
```diff
@@ -237,73 +287,81 @@ WITH null AS n1, null as n2
...
 ----
-MATCH (node:NonSpacedLabel {property: 42})
-RETURN labels(node)
+MATCH (:Person)-->(vehicle:Car)-->(:Company)<--
+      (:Country)
+RETURN count(vehicle)
 ----
 
-. Do not use a semicolon at the end of the statement.
+. Surround node patterns with parentheses.
```
Comment (petraselmer) [inline docs/style-guide.adoc]: I wonder whether we should zap this section (surrounding node patterns with parentheses) entirely? I don't think there is much value about talking about old Cypher code here...

File: docs/style-guide.adoc [hunk 2]
```diff
@@ -415,6 +473,7 @@ RETURN map
 ----
 
 . No padding space for parameters.
+.. This rule mentions deprecated syntax. See link:https://github.com/opencypher/openCypher/blob/master/cip/2.testable/CIP2016-07-07-Parameter-syntax.adoc#interaction-with-existing-features[Parameter Syntax].
```
Comment (petraselmer) [inline docs/style-guide.adoc]: In contrast to my comment about surrounding nodes with parentheses, this is one rule I would definitely keep for the time being 👍

File: docs/style-guide.adoc [hunk 3]
```diff
@@ -526,6 +585,24 @@ RETURN split( 'original', 'i' )
 RETURN split('original', 'i')
 ----
 
+. Use padding space on the inside of simple subquery expressions.
```
Comment (petraselmer) [inline docs/style-guide.adoc]: "Use padding space on the inside of simple subquery expressions."
->
"Use padding space within simple subquery expressions."

File: docs/style-guide.adoc [hunk 4]
```diff
@@ -105,120 +107,168 @@ RETURN a.prop
...
+----
++
+.Good
+[source, cypher]
+----
+MATCH (e:EditorInChief)-->(:Employee)
+RETURN e.name
+----
+
+. Write relationship types in upper case, using underscore to separate words.
```
Comment (Mats-SX) [inline docs/style-guide.adoc]: How about 

> an underscore (`_`)

File: docs/style-guide.adoc (hunk 1 above)
Comment (Mats-SX) [inline docs/style-guide.adoc]: I have been thinking the same thing on several occasions. Now you provided me the courage necessary. Will zap.

**Pull request merged**